src/Tools/dotnet-monitor/ParameterCapturing/CaptureParametersOperation.cs

```diff
@@ -0,0 +1,190 @@
+
+        private async Task<bool> CanEndpointProcessRequestsAsync(CancellationToken token)
+        {
+            DiagnosticsClient client = new(_endpointInfo.Endpoint);
+
+            IDictionary<string, string> env = await client.GetProcessEnvironmentAsync(token);
+
+            if (!env.TryGetValue(InProcessFeaturesIdentifiers.EnvironmentVariables.AvailableInfrastructure.ManagedMessaging, out string isManagedMessagingAvailable) ||
+                !env.TryGetValue(InProcessFeaturesIdentifiers.EnvironmentVariables.AvailableInfrastructure.HostingStartup, out string isHostingStartupAvailable))
+            {
+                return false;
+            }
+
+            return ToolIdentifiers.IsEnvVarValueEnabled(isManagedMessagingAvailable) && ToolIdentifiers.IsEnvVarValueEnabled(isHostingStartupAvailable);
+        }
+
+        public async Task ExecuteAsync(TaskCompletionSource<object> startCompletionSource, CancellationToken token)
+        {
+            // Check if the endpoint is capable of responding to our requests
+            if (!await CanEndpointProcessRequestsAsync(token))
+            {
+                throw new MonitoringException(Strings.ErrorMessage_ParameterCapturingNotAvailable);
+            }
+
+            EventParameterCapturingPipelineSettings settings = new()
+            {
+                Duration = Timeout.InfiniteTimeSpan
+            };
+            settings.OnStartedCapturing += OnStartedCapturing;
+            settings.OnStoppedCapturing += OnStoppedCapturing;
+            settings.OnCapturingFailed += OnCapturingFailed;
+            settings.OnServiceStateUpdate += OnServiceStateUpdate;
+            settings.OnUnknownRequestId += OnUnknownRequestId;
+
+            await using EventParameterCapturingPipeline eventTracePipeline = new(_endpointInfo.Endpoint, settings);
+            Task runPipelineTask = eventTracePipeline.StartAsync(token);
+
+            await _profilerChannel.SendMessage(
+                _endpointInfo,
+                new JsonProfilerMessage(IpcCommand.StartCapturingParameters, new StartCapturingParametersPayload
+                {
+                    RequestId = _requestId,
+                    Duration = _duration,
+                    Methods = _methods
+                }),
+                token);
+
+            try
+            {
+                await _capturingStartedCompletionSource.Task.WaitAsync(token).ConfigureAwait(false);
+                startCompletionSource?.TrySetResult(null);
+
+                await _capturingStoppedCompletionSource.Task.WaitAsync(token).ConfigureAwait(false);
+            }
+            catch (OperationCanceledException)
+            {
+                _ = _capturingStartedCompletionSource.TrySetCanceled(token);
+                _ = _capturingStoppedCompletionSource.TrySetCanceled(token);
+
+                using CancellationTokenSource stopCancellationToken = new(TimeSpan.FromSeconds(30));
+                await StopAsync(stopCancellationToken.Token);
```

Comment:
```suggestion
                await StopAsync(stopCancellationToken.Token).SafeAwait();
```
I don't think we want the ExecuteAsync method Task to contain the failure due to stopping being cancelled but rather let the original exception be the one that ultimately fails the task.